sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -116,7 +137,7 @@ case class GpuUnaryPositive(child: Expression) extends GpuUnaryExpression
 
 case class GpuAbs(child: Expression, failOnError: Boolean) extends CudfUnaryExpression
     with ExpectsInputTypes with NullIntolerant {
-  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(TypeCollection.NumericAndInterval)
```

Comment:
Do not need shims for this change ?

Reply:
Spark shimed this:
Spark 3.3  
```
val NumericAndAnsiInterval = TypeCollection(
    NumericType,
    DayTimeIntervalType,
    YearMonthIntervalType)
```
Spark 3.2
```
  val NumericAndInterval = TypeCollection(NumericType, CalendarIntervalType)
```

And `TypeCollection.NumericAndInterval` is refered many times in this file. It's safe